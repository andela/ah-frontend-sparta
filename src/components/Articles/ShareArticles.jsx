import React from 'react';
import ShareLinks from '../../helpers/socialShare';
import './article.scss';


const shareArticles = ({ article }) => {
  const articleShare = ShareLinks(article, article.slug);
  return (
    <div className="icons">
      <div className="dropup">
        <button type="button" className="btn custom-share-button  float-right" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
          Share
          {' '}
          <i className="fas fa-share-alt" />
        </button>
        <div className="dropdown-menu">
          <a className="dropdown-item" href={articleShare.facebookShare}>
            <i className="fab fa-facebook-square facebook-icon" />
            Facebook
          </a>
          <a className="dropdown-item" href={articleShare.twitterShare}>
            <i className="fab fa-twitter twitter-icon" />
            Twitter
          </a>
          <a className="dropdown-item" href={articleShare.mailshare}>
            <i className="fas fa-envelope-square envelop-icon" />
            Mail
          </a>
        </div>
      </div>
    </div>
  );
};

export default shareArticles;
