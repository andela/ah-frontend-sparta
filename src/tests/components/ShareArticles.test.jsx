import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import ShareArticles from '../../components/Articles/ShareArticles';

describe('Share Articles Component', () => {
  let mountApp;
  const props = {
    article: {
      id: 1,
      author: {
        username: 'rogha',
      },
      article_read_time: 'less than a minute read',
      title: 'Post one',
      description: 'too long a story',
      slug: 'post-one-first',
      body: 'ahhhhh --edit me',
      createdAt: '2019-04-18T11:55:58.111843Z',
      tags: ['Arsenal', 'soccer', 'Andela'],
    },
  };
  beforeEach(() => {
    mountApp = renderer.create(
      <Router>
        <ShareArticles {...props} />
      </Router>,
    );
  });

  it('Should render without crashing', () => {
    expect(mountApp).toMatchSnapshot();
  });
});
