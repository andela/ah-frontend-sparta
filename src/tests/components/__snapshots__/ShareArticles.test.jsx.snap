// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share Articles Component Should render without crashing 1`] = `
<div
  className="icons"
>
  <div
    className="dropup"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="btn custom-share-button  float-right"
      data-toggle="dropdown"
      type="button"
    >
      Share
       
      <i
        className="fas fa-share-alt"
      />
    </button>
    <div
      className="dropdown-menu"
    >
      <a
        className="dropdown-item"
        href="https://www.facebook.com/sharer/sharer.php?u=undefined/articles/post-one-first"
      >
        <i
          className="fab fa-facebook-square facebook-icon"
        />
        Facebook
      </a>
      <a
        className="dropdown-item"
        href="https://twitter.com/intent/tweet?url=undefined/articles/post-one-first&text=too long a story"
      >
        <i
          className="fab fa-twitter twitter-icon"
        />
        Twitter
      </a>
      <a
        className="dropdown-item"
        href="mailto:?Subject=Post%20one&body=I found this article to be worth reading, Post one undefined/articles/post-one-first"
      >
        <i
          className="fas fa-envelope-square envelop-icon"
        />
        Mail
      </a>
    </div>
  </div>
</div>
`;
